rasa_nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -379,53 +395,81 @@ def _create_batch_b(self, batch_pos_b, intent_ids):
+                                 int(len(X) % batch_size > 0))
+
+            ep_loss = 0
             for i in range(batches_per_epoch):
-                end_idx = (i + 1) * self.batch_size
-                start_idx = i * self.batch_size
+                end_idx = (i + 1) * batch_size
+                start_idx = i * batch_size
                 batch_a = X[indices[start_idx:end_idx]]
                 batch_pos_b = Y[indices[start_idx:end_idx]]
                 intents_for_b = intents_for_X[indices[start_idx:end_idx]]
                 # add negatives
                 batch_b = self._create_batch_b(batch_pos_b, intents_for_b)
 
-                sess_out = sess.run({'loss': loss, 'train_op': train_op},
-                                    feed_dict={a_in: batch_a,
-                                               b_in: batch_b,
-                                               is_training: True})
-
-            if logger.isEnabledFor(logging.INFO) and (ep + 1) % 10 == 0:
-                self._output_training_stat(X, intents_for_X, all_Y,
-                                           sess, a_in, b_in,
-                                           sim, is_training,
-                                           ep, sess_out)
-
-    def _output_training_stat(self,
-                              X, intents_for_X, all_Y,
-                              sess, a_in, b_in, sim, is_training,
-                              ep, sess_out):
+                sess_out = self.session.run(
+                        {'loss': loss, 'train_op': train_op},
+                        feed_dict={self.a_in: batch_a,
+                                   self.b_in: batch_b,
+                                   is_training: True}
+                )
+                ep_loss += sess_out.get('loss') / batches_per_epoch
+
+            if self.calc_acc_on_num_examples:
+                if ((ep + 1) == 1 or
+                        (ep + 1) % self.calc_acc_ones_in_epochs == 0 or
+                        (ep + 1) == self.epochs):
+                    train_acc = self._output_training_stat(X, intents_for_X,
+                                                           is_training)
+                    last_loss = ep_loss
+
+                pbar.set_postfix({
+                    "loss": "{:.3f}".format(ep_loss),
+                    "acc": "{:.3f}".format(train_acc)
+                })
+            else:
+                pbar.set_postfix({
+                    "loss": "{:.3f}".format(ep_loss)
+                })
+
+        if self.calc_acc_on_num_examples:
+            logger.info("Finished training embedding policy, "
+                        "loss={:.3f}, train accuracy={:.3f}"
+                        "".format(last_loss, train_acc))
+
+    def _output_training_stat(self, X, intents_for_X, is_training):
         """Output training statistics"""
+        n = self.calc_acc_on_num_examples
+        ids = np.random.permutation(len(X))[:n]
+        all_Y = self._create_all_Y(X[ids].shape[0])
```

Comment:
why do we actually need to do this? Doesn't tensorflow automatically evaluate on the training data if we tell it to do so or is this coming from keras? Because when training the keras model this is displayed as well

Reply:
I think it is coming from keras

Reply:
ah good to know :+1: